### PR TITLE
[14.0][OU-ADD] sale_margin: Migration scripts

### PR DESCRIPTION
**Pull request by hieu1211**
This PR
======
migration done

### CODE REVIEW

File: docsource/modules130-140.rst [hunk 1]
```diff
@@ -596,7 +596,7 @@ Module coverage 13.0 -> 14.0
 +--------------------------------------------+-------------------------------------------------+
 |sale_management                             | Done                                            |
 +--------------------------------------------+-------------------------------------------------+
-|sale_margin                                 |                                                 |
+|sale_margin                                 | Done                                               |
```
Comment (MiquelRForgeFlow) [inline docsource/modules130-140.rst]:
```suggestion
|sale_margin                                 | Done                                            |
```
to align  |

File: openupgrade_scripts/scripts/sale_margin/14.0.1.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+@openupgrade.migrate()
+def migrate(env, installed_version):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN margin_percent float;
+
+        ALTER TABLE sale_order
+        ADD COLUMN margin_percent float;
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/sale_margin/14.0.1.0/pre-migration.py]:
```suggestion
        ADD COLUMN margin_percent numeric;

        ALTER TABLE sale_order
        ADD COLUMN margin_percent numeric;
```

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by tranngocson1996 — APPROVED** [review]

**Comment by hieu1211** [comment]
@MiquelRForgeFlow i changed following u but the type column in DB is double_precision, it that correct?

**Comment by MiquelRForgeFlow** [comment]
@hieu1211 you can find the corresponding type in https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L2309.

Selection_150

**Comment by hieu1211** [comment]
> @hieu1211 you can find the corresponding type in https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L2309.
> 
> Selection_150

thanks for responding, i will check it

**Comment by royle-vietnam** [comment]
@MiquelRForgeFlow maybe you are wrong
Field `margin_percent` that is not digits, i think `double precision` is more correct than `numeric`
```python
margin_percent = fields.Float(
        "Margin (%)", compute='_compute_margin', store=True, groups="base.group_user")
```

https://github.com/OCA/openupgradelib/blob/a8e94ce510e408c55596827d38a9dc80a926433a/openupgradelib/openupgrade.py#L2315
```python
'float': 'numeric',  # Force manually to double precision if no digits
```

**Comment by MiquelRForgeFlow** [comment]
Oh, right, sorry for the noise.

**Review by pedrobaeza — APPROVED** [review]
FYI, the openupgradelib method was develop from reading this line of Odoo code:

https://github.com/odoo/odoo/blob/098f00724917f36c60b4850da5105556239edd1e/odoo/fields.py#L1314

**Pull request merged**